### PR TITLE
added required services, annotations, etc.

### PR DESCRIPTION
**Pull request by AndriiBardin**

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/taxi/Main.java [hunk 1]
```diff
@@ -10,25 +14,79 @@
...
 
         System.out.println(manufacturerService.getAll());
 
+        driverService.create(one);
+        driverService.create(two);
+        driverService.create(three);
+        driverService.create(four);
+        driverService.create(five);
+
+        Driver driverUpdate = driverService.update(two);
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/Main.java:63]: are any changed applied here? did you mean to get() the driver instead?

File: src/main/java/taxi/Main.java [hunk 2]
```diff
@@ -10,25 +14,79 @@
...
+        driverUpdate.setLicenceNumber("39800");
+        System.out.println(driverService.get(2L));
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/Main.java:65]:
```suggestion
        System.out.println(driverService.update(driverUpdate));
```
let's update the driver after we altered some information about him
same for manufacturer

File: src/main/java/taxi/Main.java [hunk 3]
```diff
@@ -10,25 +14,79 @@
...
         System.out.println(manufacturerService.getAll());
 
-        manufacturerService.delete(3L);
-
-        System.out.println(manufacturerService.getAll());
+        manufacturerService.delete(5L);
 
         Manufacturer manufacturerUpdate = manufacturerService.get(2L);
-        manufacturerUpdate.setCountry("Japan, Minato City");
+        manufacturerUpdate.setCountry("Japan, Minato-City");
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/Main.java:53]: let's call update() after we changed the country

File: src/main/java/taxi/dao/CarDao.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package taxi.dao;
+
+import java.util.List;
+import java.util.Optional;
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/CarDao.java:5]: redundant empty line

File: src/main/java/taxi/dao/CarDao.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import taxi.lib.Dao;
+import taxi.model.Car;
+
+@Dao
+public interface CarDao {
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/CarDao.java:10]: do we need this annotation here?

File: src/main/java/taxi/dao/CarDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public Car create(Car car) {
+        Storage.add(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        return getAll().stream()
+                .filter(car -> Objects.equals(car.getId(),id))
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/CarDaoImpl.java:24]: can we just use `equals()`?

File: src/main/java/taxi/dao/CarDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        IntStream.range(0, Storage.cars.size())
+                .filter(i -> Storage.cars.get(i).getId().equals(car.getId()))
+                .forEach(i -> Storage.cars.set(i, car));
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.cars
+                .removeIf(i -> Objects.equals(i.getId(), id));
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/CarDaoImpl.java:44]: same

File: src/main/java/taxi/dao/DriverDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
+package taxi.dao;
+
+import taxi.db.Storage;
+import taxi.lib.Dao;
+import taxi.model.Driver;
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/DriverDaoImpl.java:6]: redundant empty line

File: src/main/java/taxi/dao/DriverDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    @Override
+    public Driver create(Driver driver) {
+        Storage.add(driver);
+        return driver;
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        return getAll().stream()
+                .filter(driver -> Objects.equals(driver.getId(),id))
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/DriverDaoImpl.java:23]:
```suggestion
                .filter(driver -> driver.getId().equals(id))
```

File: src/main/java/taxi/dao/DriverDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        IntStream.range(0, Storage.drivers.size())
+                .filter(i -> Storage.drivers.get(i).getId().equals(driver.getId()))
+                .forEach(i -> Storage.drivers.set(i, driver));
+        return driver;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.drivers
+                .removeIf(i -> Objects.equals(i.getId(), id));
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/DriverDaoImpl.java:43]: same

File: src/main/java/taxi/db/Storage.java [hunk 11]
```diff
@@ -2,14 +2,31 @@
 
 import java.util.ArrayList;
 import java.util.List;
+
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/db/Storage.java:5]: redundant empty line
you should run checkstyle before push

File: src/main/java/taxi/model/Car.java [hunk 12]
```diff
@@ -46,4 +46,19 @@ public Manufacturer getManufacturer() {
...
+                + "id="
+                + id
+                + ", model='"
+                + model
+                + '\''
+                + ", manufacturer="
+                + manufacturer
+                + ", drivers="
+                + drivers
+                + '}';
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/model/Car.java:62]:
```suggestion
        return "Car {"
                + "id=" + id
                + ", model='" + model
                + ", manufacturer=" + manufacturer
                + ", drivers=" + drivers + '}';
```
let's make this more readable

File: src/main/java/taxi/model/Driver.java [hunk 13]
```diff
@@ -33,4 +33,18 @@ public String getName() {
...
+        return "Driver{"
+                + "id="
+                + id
+                + ", name='"
+                + name
+                + '\''
+                + ", licenceNumber='"
+                + licenceNumber
+                + '\''
+                + '}';
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/model/Driver.java:48]: same

File: src/main/java/taxi/services/CarService.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.List;
+import taxi.model.Car;
+import taxi.model.Driver;
+
+public interface CarService {
+    Car create(Car car);
+
+    Car update(Car car);
+
+    Car get(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/services/CarService.java:12]: getAll() method is missing

File: src/main/java/taxi/dao/CarDao.java [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+
+    Optional<Car> get(Long id);
+
+    List<Car> getAll();
+
+    Car update(Car car);
+
+    boolean delete(Long id);
+
+    List<Car> getAllById(Long driverId);
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/CarDao.java:21]:
```suggestion
    List<Car> getAllByDriverId(Long driverId);
```

File: src/main/java/taxi/services/DriverServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public Driver create(Driver driver) {
+        return driverDao.create(driver);
+    }
+
+    @Override
+    public Driver get(Long id) {
+        return driverDao.get(id).orElseThrow(
+                () -> new RuntimeException("Car with id " + id + " not found"));
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/services/DriverServiceImpl.java:23]: why Car?

**Review by KseniiaMakarova — APPROVED** [review]
fix comments before merging

**Pull request merged**